### PR TITLE
Feature/install plugins

### PR DESCRIPTION
**Pull request by EugenMayer**
Be aware, this patch  is based on pull request #49 since it already uses the boostrap-scripts.

I would be easy to detach this or even include the script inline - but i would keep them outside

### CODE REVIEW

**Comment by ghost** [comment]
[CLA assistant check](https://cla.gocd.org/gocd/docker-gocd-server?pullRequest=51) <br/>All committers have signed the CLA.

File: Dockerfile [hunk 1]
```diff
@@ -44,10 +44,12 @@ RUN \
...
   unzip /tmp/go-server.zip -d / && \
   rm /tmp/go-server.zip && \
-  mv go-server-17.11.0 /go-server
+  mv go-server-17.11.0 /go-server && \
+  mkdir -p /docker-entrypoint.d
 
 COPY logback-include.xml /go-server/config/logback-include.xml
 
 ADD docker-entrypoint.sh /
+ADD install_plugins.sh /docker-entrypoint.d/install_plugins.sh
```
Comment (ketan) [inline Dockerfile:53]: You should not touch this file. It's generated automatically via a build script.

File: install_plugins.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    plugin_dest=${GOCD_PLUGIN_EXTERNAL_HOME}/${plugin_filename}
+    if [ ! -f plugin_dest ]; then
+        echo "installing plugin $plugin_name from $plugin_url to $plugin_dest"
+        # TODO: should we use silent?
+        curl --silent --location --retry 3 ${plugin_url} -o ${plugin_dest}
+    else
+        echo "installing plugin ${plugin_name} already installed"
+    fi
+  fi
+done < <(env)
```
Comment (ketan) [inline install_plugins.sh:25]: This feels icky. Can we provide some form of structured input to the script, probably by writing out some "structured" data to a known location?

```
# /gocd-plugin-list
PLUGINS=(
  "docker:https://github.com/...",
  "k8s:https://github.com/..."
)
```

https://stackoverflow.com/questions/688849/associative-arrays-in-shell-scripts/4444841#4444841

**Comment by EugenMayer** [comment]
this makes all this unflexible und people will fork again ( IMHO ). The point is, getting those things whereever they are, from my repo, you repo, my hidden repo. This is technical, and anybody would do is a technican. 

I would not go and design this as easy to install wizard. Better technical concept to deal with the usual installation tasks and shortcumings when handling with docker images.

My cents

**Comment by EugenMayer** [comment]
Ah wait, we misunderstood eachother. Yours would need adjusting the image or at least add a custom script, mine is customizeable using ENV vars only, so can be run straight in a orchestrator out of the box.

There are no "arrays" in ENV, thats why NAMESPACE_<key>=value is the most common thing i see here. Be it for nginx/vhosts/lets encrypt tuples or others.

Again that is what i know, what i have seen most, used most and adopted as its practical - so people might recognize that pattern from other projects too.

If ENV could make up with some arrays (or rather hahsmaps), thing thing could be a lot nicer - but thats not included and not included in docker-compose.yml ( there is an issue for that, its open ).

I am running this in my image now, works well.

**Comment by ketan** [comment]
Closing in favor of #54. Copied your script and tweaked it a bit so that `/docker-entrypoint.d` itself can be volume mounted.

**Comment by EugenMayer** [comment]
a very odd way to go with contributions .. anyway, at least it is part of gocd. Thanks.

**Pull request closed without merging**